### DISCUSSION TITLE
Error on gitlog_showcase

### DISCUSSION THREAD

**Discussion by ian-lastname** [sailuh/kaiaulu#265] [Q&A] [answered]
I try to run prase_gitlog(perceval_path,git_repo_path) and it gives me an error "Error in system2("git", args = c("--git-dir", git_repo_path, "--work-tree", : error in running command"

**Comment by carlosparadis** [comment]
We discussed on call the error message is being caused due to the data being analyzed not being used (in this case, the git log of the project APR). Ian, to follow-up on this as we agreed, take a look on the Notebook narrative in english and see if what project configuration files are, and what data is needed to run the code block. Then try to run parse_gitlog. It seems you figured the tools.yml already too. 

If not, please follow up here with a reply where it is not clear. For now, a TODO I have is to fix the error message when data is missing, since it is outright cryptic. I believe the error message for when Perceval is missing was added, but I overlook the data.

For the User Experience Report, please refer to this item mentioning the fact the expectation was the code to run, without the APR data at a first step, since I think unfamiliarity with Mining Software Repositories may have played a role on this, which is totally fine.  👍

**Comment by carlosparadis** [comment]
Possibly duplicate with https://github.com/sailuh/kaiaulu/discussions/269

**Comment by ian-lastname** [comment]
I tried adding the path that was given in apr.yml into my files ("../../rawdata/git_repo/APR/.git"), I'm still getting the "Error in system2("git", args = c("--git-dir", git_repo_path, "--work-tree", : error in running command" error. I tried reading the english narrative, which is what led me to manually creating the path.

**Reply by carlosparadis** [reply]
I need more context to understand what is going on here. Did you download the APR data and place it in the location specified? Do you want understand the project configuration file and what the Notebook is trying to do?

**Reply by ian-lastname** [reply]
I installed the apr repo from the git_url link, though I can’t find any .git file in what I got. Am I downloading the correct thing, or am I completely off?

**Reply by carlosparadis** [reply]
The prefixed `.` in `.`git means it is an invisible folder. Are you going via the terminal and not able to enter the folder? Does it say it doesn't exist?

**Reply by ian-lastname** [reply]
Yes, I checked in my terminal too. The closest thing is .gitignore and .github

**Reply by carlosparadis** [reply]
Can you describe me the steps which you do to download APR? Maybe you are downloading the source code but not the version control.

**Reply by ian-lastname** [reply]
Followed the link to the GitHub, downloaded via the green code button on the top right. I think you’re right, I’m just downloading the source code. How do I download the version control?

**Reply by carlosparadis** [reply, marked as answer]
Can you try using git clone on the terminal and the url it provides on green button?

**Reply by ian-lastname** [reply]
Alright, that worked. Thank you very much.

**Reply by carlosparadis** [reply]
@ian-lastname Ok! Could you confirm then the issue was you were downloading the source code, without the .git, and therefore the error? If yes, please go ahead and mark my last message as answer. If you have other questions on the git log notebook that doesn't run, please open a new one so the thread doesn't grow too long.

**Reply by ian-lastname** [reply]
I can confirm. It was simply that the .git file was not in there.
